### ISSUE TITLE
No component factory found for X after upgrade to Angular 5.1.0 with AOT build

### ISSUE THREAD

**Issue by jozsefm** [angular/angular#20945]
## I'm submitting a...

<pre><code>
[x] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

After upgrading to version 5.1.0 from 4.4.6 and tweaked config to successfully emit AOT build our app started to throw this error:
<code>angular.js:12477 Error: No component factory found for TestPage. Did you add it to @NgModule.entryComponents? at noComponentFactoryError (file:///D:/angular5ngFactoryError/angular-js/app/app.v5.js:4181:34) at CodegenComponentFactoryResolver.resolveComponentFactory (file:///D:/angular5ngFactoryError/angular-js/app/app.v5.js:4245:19) at doDowngrade (file:///D:/angular5ngFactoryError/angular-js/app/app.v5.js:23770:70) at link (file:///D:/angular5ngFactoryError/angular-js/app/app.v5.js:23803:21) at invokeLinkFn (file:///D:/angular5ngFactoryError/angular-js/node_modules/angular/angular.js:8789:9) at nodeLinkFn (file:///D:/angular5ngFactoryError/angular-js/node_modules/angular/angular.js:8289:11) at compositeLinkFn (file:///D:/angular5ngFactoryError/angular-js/node_modules/angular/angular.js:7680:13) at compositeLinkFn (file:///D:/angular5ngFactoryError/angular-js/node_modules/angular/angular.js:7684:13) at publicLinkFn (file:///D:/angular5ngFactoryError/angular-js/node_modules/angular/angular.js:7555:30) at file:///D:/angular5ngFactoryError/angular-js/node_modules/angular/angular.js:1662:27 </code>

This prevents the app from starting. As you can see this happens in a hybrid setup where I try to use a downgraded Angular component in an AngularJS app from an umd file built with Webpack.
## Expected behavior
The built version should work just fine as before, as no code has changed, only the framework.

## Minimal reproduction of the problem with instructions

git clone https://github.com/jozsefm/angular5ngFactoryError.git
follow steps written in the readme: https://github.com/jozsefm/angular5ngFactoryError 
basically you only need to open the index.html

## What is the motivation / use case for changing the behavior?

I would like to use the latest version of Angular in our AngularJS app.

## Environment

<pre><code>
Angular version: 5.1.0
AngularJS version: 1.4.7

Browser:
- [X] Chrome (desktop) version 62
 
For Tooling issues:
- Node version: 6.11.4 
- Platform:  Windows 

Others:

</code></pre>

**Comment by chuckjaz** [comment]
This might be related to or a duplicate of #20091

**Comment by jozsefm** [comment]
I'm not sure if I understand #20091 correctly, but in my case, there is no build error while executing ngc, the ngfactory.js files get generated. If you check the output of webpack the contents of the factory are inside the final umd package. So I think this is a different scenario but again I might just miss the connection.

**Comment by chuckjaz** [comment]
@jozsefm I am having trouble reproducing this.

I followed the steps in README.md file in the root the directory and I receive the following error in the `angular-4` and `angular-5`: 

```
ERROR in [copy-webpack-plugin] unable to locate './dist' at '/Users/chuckj/bugs/20945/angular5ngFactoryError/angular-4/dist'
```

This appears to be from webpack and not ngc.

If execute just `npm run ngc` in both directories and neither produce an error.

**Comment by jozsefm** [comment]
Yeah that error is caused by the CopyWebpackPlugin step only for the first time for some reasons I don't totally get. The output files get generated correctly besides that and it goes away after the first time. 
The real error I reported happens during application runtime, after you open the index.html. I included a pre-built version of both repos for testing so running webpack is only necessary to test the build. I will update the description to emphasize this better and mention the error.
The reason for the copy step is just to make the process a little easier by eliminating the manual copying of files, if there is an error because it fails, that is not effecting the core issue itself.

**Comment by Elliottssu** [comment]
I have the same problem, and I'm looking forward to fixing it. its very important. Our company's products are going online in only two weeks, and if it don't have AOT, the experience is very bad.

**Comment by Delagen** [comment]
Any updates? Released 5.2.0. Seems only 5.1.1 version works now.

**Comment by jozsefm** [comment]
I have checked the reproduction repo using version 5.1.1 and 5.2.0 and for me the issue is still present, so I think @Delagen we might not facing the same problem. I hope @chuckjaz or someone will take a second look as I updated to repo to run the build error free.

**Comment by gaelmigliorini** [comment]
Same problem for me : 
`main.04dd2d1a7b6b3defdbb0.bundle.js:1 ERROR Error: No component factory found for function (){this.message=e}. Did you add it to @NgModule.entryComponents?
    at ge (main.04dd2d1a7b6b3defdbb0.bundle.js:1)
    at t.LMZF.t.resolveComponentFactory (main.04dd2d1a7b6b3defdbb0.bundle.js:1)
    at t.Un6q.t.ngOnChanges (main.04dd2d1a7b6b3defdbb0.bundle.js:1)
    at main.04dd2d1a7b6b3defdbb0.bundle.js:1
    at zo (main.04dd2d1a7b6b3defdbb0.bundle.js:1)
    at fs (main.04dd2d1a7b6b3defdbb0.bundle.js:1)
    at Object.updateDirectives (main.04dd2d1a7b6b3defdbb0.bundle.js:1)
    at Object.updateDirectives (main.04dd2d1a7b6b3defdbb0.bundle.js:1)
    at Uo (main.04dd2d1a7b6b3defdbb0.bundle.js:1)
    at Yo (main.04dd2d1a7b6b3defdbb0.bundle.js:1)`

Waiting for updates

**Comment by mlc-mlapis** [comment]
@gaelmigliorini ... hmm, any really simple reproduction to be able to run locally via ngc? If not possible to reproduce by that way then it is CLI case for sure ... and not Angular at all.

**Comment by jozsefm** [comment]
@mlc-mlapis I do not use the cli in my case only webpack directly. So this can not be CLI related in my setup. Also it is not happening during runing ngc or webpack but only when you try to run the built app.

**Comment by gaelmigliorini** [comment]
@mlc-mlapis @jozsefm I'm using the CLI but this problem also occurs in the console when running the app, there's no error during ng build --aot.

**Comment by patrickmichalina** [comment]
I seem to be having the same issue using Material Spinner when used as an entry component but only in AOT builds...

**Comment by alxhub** [comment]
@jozsefm apologies for the long delay on this. The pre-built bundles in the repro work but I'm unable to run the builds myself, or to test with the latest Angular. I get this error:

```
Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration.module has an unknown property 'loaders'. These properties are valid:
   object { defaultRules?, exprContextCritical?, exprContextRecursive?, exprContextRegExp?, exprContextRequest?, noParse?, rules?, strictExportPresence?, strictThisContextOnImports?, unknownContextCritical?, unknownContextRecursive?, unknownContextRegExp?, unknownContextRequest?, unsafeCache?, wrappedContextCritical?, wrappedContextRecursive?, wrappedContextRegExp? }
   -> Options affecting the normal modules (`NormalModuleFactory`).
```

I'm going to tag this as "needs repro" for now. Please feel free to update the repro if this is still an issue for you, and I'll look into it.

**Comment by alxhub** [comment]
Closing this due to inactivity; feel free to open a new issue with a reproduction if the problem is still occurring.
